### PR TITLE
E2E tests for examples/rethinkdb

### PR DESCRIPTION
**Pull request by mwielgus**

### CODE REVIEW

**Comment by vmarmol** [comment]
Assigning to @quinton-hoole, feel free to reassign :)

**Comment by mwielgus** [comment]
ref: #10013

**Comment by erictune** [comment]
judgement call: adding tests at this point is more risk of flake than chance of catch a bug.  therefore, move to v1-post.  if you did find a bug when writing this test, please call that out and we can decide if test + fix merits v1.

File: test/e2e/examples.go [hunk 1]
```diff
@@ -157,6 +157,47 @@ var _ = Describe("Examples e2e", func() {
...
+			checkDbInstances := func() {
+				forEachPod(c, ns, "db", "rethinkdb", func(pod api.Pod) {
+					_, err := lookForStringInLog(ns, pod.Name, "rethinkdb", "Server ready", serverStartTimeout)
+					Expect(err).NotTo(HaveOccurred())
+				})
+			}
+			checkDbInstances()
+
+			By("scaling rethinkdb")
+			ScaleRC(c, ns, "rethinkdb-rc", 2)
```
Comment (gmarek) [inline test/e2e/examples.go]: I think it's OK to leave it as it is, but I'm slightly worried that hardcoding number of replicas here may cause some problems in the future. If someone would change the default replica count in the example, tests won't catch this change. It'd be cleaner to have an util method "AddOneReplica", or check the replica count beforehand.

**Comment by gmarek** [comment]
LGTM

**Comment by gmarek** [comment]
@k8s-bot please test this

**Comment by vishh** [comment]
@k8s-bot ok to test

**Pull request merged**